core/src/main/scala/org/http4s/Message.scala

```diff
@@ -337,14 +337,21 @@ sealed trait MaybeResponse {
 }
 
 object MaybeResponse {
-  implicit val instance: Monoid[MaybeResponse] =
+  /** Intentionally not implicit. Do not use unless you've read and understood the
+    * deprecation warning on [[instance]].
+    */
+  val monoidInstance: Monoid[MaybeResponse] =
     new Monoid[MaybeResponse] {
       def zero =
         Pass
       def append(a: MaybeResponse, b: => MaybeResponse) =
         a orElse b
     }
 
+  @deprecated("""We derive `Monoid[Kleisli[Task, Request, MaybeResponse]]` from `Monoid[Task[MaybeResponse]]`, which needs to have the "orElse"-like semantics of `taskInstance`.  The presence of this monoid makes it easy to locate a monoid that evaluates both effects, which is not what we want when chaining `HttpService`s.  Use `taskInstance` instead.""", "0.16.3")
```

Comment:
I like this message. Although I'm not sure it screams I may run two separate effects and only return the first and ignore the second entirely. Also breaking resource safety in the process possibly.

Reply:
I hope people have fatal warnings turned on and _that_ screams at them.  But we could make this more stearn, I suppose.